In [1]:
#!jupyter notebook --NotebookApp.iopub_data_rate_limit=1.0e10
import csv

In [6]:
subDict = dict()

for i in range(12):
    with open("data/submissions/csv-" + str(i) + ".csv", mode='r', encoding="Latin1") as subFile:
        #print(i)
        reader = csv.DictReader(subFile)
        for row in reader:
            if row["subreddit_id"] in subDict:
                subDict[row["subreddit_id"]]["numOfSubmissions"] += 1
                if row["author"] not in subDict[row["subreddit_id"]]["users"]:
                    subDict[row["subreddit_id"]]["users"].append(row["author"])
            else:
                subDict[row["subreddit_id"]] = {
                    "subreddit": row["subreddit"],
                    "numOfSubmissions": 1,
                    "users": [row["author"]],
                    "numOfComments": 0
                }
    
    with open("data/comments/csv-" + str(i) + ".csv", mode='r', encoding="Latin1") as comFile:
        print(i)
        reader = csv.DictReader(comFile)
        for row in reader:
            if row["subreddit_id"] in subDict:
                subDict[row["subreddit_id"]]["numOfComments"] += 1
                if row["author"] not in subDict[row["subreddit_id"]]["users"]:
                    subDict[row["subreddit_id"]]["users"].append(row["author"])
            else:
                subDict[row["subreddit_id"]] = {
                    "subreddit": row["subreddit"],
                    "numOfSubmissions": 0,
                    "users": [row["author"]],
                    "numOfComments": 1
                }

0
1
2
3
4
5
6
7
8
9
10
11


In [15]:
usersDict = dict()

for i in range(12):
    with open("data/submissions/csv-" + str(i) + ".csv", mode='r', encoding="Latin1") as subFile:
        #print(i)
        reader = csv.DictReader(subFile)
        for row in reader:
            if row["author"] == "[deleted]":
                continue
            if row["author"] in usersDict:
                usersDict[row["author"]]["numOfSubmissions"] += 1
                if row["subreddit_id"] not in usersDict[row["author"]]["subreddit_ids"]:
                    usersDict[row["author"]]["subreddit_ids"].append(row["subreddit_id"])
            else:
                usersDict[row["author"]] = {
                    "numOfSubmissions": 1,
                    "numOfComments": 0,
                    "subreddit_ids": [row["subreddit_id"]]
                }
    
    with open("data/comments/csv-" + str(i) + ".csv", mode='r', encoding="Latin1") as comFile:
        print(i)
        reader = csv.DictReader(comFile)
        for row in reader:
            if row["author"] == "[deleted]":
                continue
            if row["author"] in usersDict:
                usersDict[row["author"]]["numOfComments"] += 1
                if row["subreddit_id"] not in usersDict[row["author"]]["subreddit_ids"]:
                    usersDict[row["author"]]["subreddit_ids"].append(row["subreddit_id"])
            else:
                usersDict[row["author"]] = {
                    "numOfSubmissions": 0,
                    "numOfComments": 1,
                    "subreddit_ids": [row["subreddit_id"]]
                }

0
1
2
3
4
5
6
7
8
9
10
11


In [7]:
print("Count: " + str(len(subDict)))
print("\nTop10 by number of submissions:")
for elem in sorted(subDict.items(), key = lambda item: item[1]["numOfSubmissions"], reverse=True)[:10]:
    print(str(elem[1]["subreddit"]) + " - " + str(elem[1]["numOfSubmissions"]))
print("\nTop10 by number of users:")
for elem in sorted(subDict.items(), key = lambda item: len(item[1]["users"]), reverse = True)[:10]:
    print(str(elem[1]["subreddit"]) + " - " + str(len(elem[1]["users"])))
print("\nTop10 by number of comments:")
for elem in sorted(subDict.items(), key = lambda item: item[1]["numOfComments"], reverse=True)[:10]:
    print(str(elem[1]["subreddit"]) + " - " + str(elem[1]["numOfComments"]))

Count: 5032

Top10 by number of submissions:
reddit.com - 1225827
politics - 253527
entertainment - 84598
pics - 82333
business - 67525
technology - 65576
worldnews - 63622
funny - 62392
science - 53171
WTF - 45212

Top10 by number of users:
reddit.com - 163780
politics - 38375
pics - 29754
technology - 28338
funny - 28187
entertainment - 26361
science - 25855
programming - 25820
business - 25254
worldnews - 24938

Top10 by number of comments:
reddit.com - 1884629
politics - 1136188
programming - 449840
pics - 414966
science - 333795
worldnews - 333123
WTF - 279226
funny - 257333
technology - 222590
AskReddit - 214145


Broj subreddit-a: 4968 (5032 ako se broje subreddit-i na kojima je bilo komentara u 2008. godini, ali ne i novih objava)

Top 10 subreddit-a po broju objeva:
1. reddit.com (/r/All) - 1225827
2. politics - 253527
3. entertainment - 84598
4. pics - 82333
5. business - 67525
6. technology - 65576
7. worldnews - 63622
8. funny - 62392
9. science - 53171
10. WTF - 45212

Top 10 subreddit-a po broju korisnika:
1. reddit.com (/r/All) - 163780
2. politics - 38375
3. pics - 29754
4. technology - 28338
5. funny - 28187
6. entertainment - 26361
7. science - 25855
8. programming - 25820
9. business - 25254
10. worldnews - 24938

Top 10 subreddit-a po broju komentara:
1. reddit.com (/r/All) - 1884629
2. politics - 1136188
3. programming - 449840
4. pics - 414966
5. science - 333795
6. worldnews - 333123
7. WTF - 279226
8. funny - 257333
9. technology - 222590
10. AskReddit - 214145

In [8]:
numOfUsers = 0
numOfSubreddits = 0
for elem in sorted(subDict.items(), key = lambda item: len(item[1]["users"]), reverse = True)[:10]:
    numOfUsers += len(elem[1]["users"])
    numOfSubreddits += 1
print(f"Prosecan broj korisnika po sabreditu: " + str(round(numOfUsers / numOfSubreddits, 2)))

Prosecan broj korisnika po sabreditu: 41666.2


In [16]:
print("\nTop10 users by number of submissions:")
for elem in sorted(usersDict.items(), key = lambda item: item[1]["numOfSubmissions"], reverse=True)[:10]:
    print(str(elem[0]) + " - " + str(elem[1]["numOfSubmissions"]))
print("\nTop10 users by number of comments:")
for elem in sorted(usersDict.items(), key = lambda item: item[1]["numOfComments"], reverse=True)[:10]:
    print(str(elem[0]) + " - " + str(elem[1]["numOfComments"]))


Top10 users by number of submissions:
gst - 18870
qgyh2 - 12238
rmuser - 9822
twolf1 - 8597
IAmperfectlyCalm - 8308
qazamisan - 6927
charlatan - 5998
igeldard - 5373
noname99 - 5334
democracy101 - 5332

Top10 users by number of comments:
NoMoreNicksLeft - 13480
malcontent - 12159
matts2 - 11672
mexicodoug - 9169
7oby - 9161
aletoledo - 8085
mutatron - 7771
otakucode - 7759
redditcensoredme - 7468
h0dg3s - 7439


Top 10 korisnika po broju objava:
1. gst - 18870
2. qgyh2 - 12238
3. rmuser - 9822
4. twolf1 - 8597
5. IAmperfectlyCalm - 8308
6. qazamisan - 6927
7. charlatan - 5998
8. igeldard - 5373
9. noname99 - 5334
10. democracy101 - 5332

Top 10 korisnika po broju komentara:
1. NoMoreNicksLeft - 13480
2. malcontent - 12159
3. matts2 - 11672
4. mexicodoug - 9169
5. 7oby - 9161
6. aletoledo - 8085
7. mutatron - 7771
8. otakucode - 7759
9. redditcensoredme - 7468
10. h0dg3s - 7439

In [18]:
print("\nTop10 users by activity:")
for elem in sorted(usersDict.items(), key = lambda item: len(item[1]["subreddit_ids"]), reverse=True)[:10]:
    print(str(elem[0]) + " - " + str(len(elem[1]["subreddit_ids"])))


Top10 users by activity:
MrKlaatu - 181
Escafane - 154
omfgninja - 122
codepoet - 111
scientologist2 - 111
turkourjurbs - 110
b34nz - 107
Sylveran-01 - 107
krugerlive - 106
tuoder - 103


Top 10 korisnika po broju na koliko su subreddit-a aktivni:
1. MrKlaatu - 181
2. Escafane - 154
3. omfgninja - 122
4. codepoet - 111
5. scientologist2 - 111
6. turkourjurbs - 110
7. b34nz - 107
8. Sylveran-01 - 107
9. krugerlive - 106
10. tuoder - 103

In [22]:
for id in usersDict["otakucode"]["subreddit_ids"]:
    print(subDict[id]["subreddit"])

reddit.com
politics
programming
science
pics
gaming
entertainment
obama
nsfw
business
worldnews
Marijuana
scifi
yourweek
bestof
technology
funny
writing
sex
WTF


In [24]:
submissionDict = dict()

for i in range(12):
    with open("data/submissions/csv-" + str(i) + ".csv", mode='r', encoding="Latin1") as subFile:
        print(i)
        reader = csv.DictReader(subFile)
        for row in reader:
            submissionDict[row["id"]] = {
                "url": row["url"],
                "permalink": row["permalink"],
                "author": row["author"],
                "created": row["created_utc"],
                "subreddit": row["subreddit"],
                "numOfComments": row["num_comments"],
                "score": row["score"],
                "over18": row["over_18"],
                "hideScore": row["hide_score"],
                "domain": row["domain"]
            }

0
1
2
3
4
5
6
7
8
9
10
11


In [31]:
print("\nTop10 submissions by number of comments:")
for elem in sorted(submissionDict.items(), key = lambda item: int(item[1]["numOfComments"]), reverse=True)[:10]:
    print(str(elem[0]) + " - " + str(elem[1]["subreddit"]) + " - " + str(elem[1]["numOfComments"]))
    if elem[1]["over18"] == "False":
        print(str(elem[1]["permalink"]))
        print(str(elem[1]["url"]))
    if elem[1]["hideScore"] == "False":
        print(str(elem[1]["score"]))


Top10 submissions by number of comments:
6nz1k - science - 33329
/r/science/comments/6nz1k/got_six_weeks_try_the_hundred_push_ups_training/
http://hundredpushups.com
1621
78n1v - WTF - 3657
/r/WTF/comments/78n1v/a_black_community_in_oh_goes_50_years_without/
http://www.time.com/time/nation/article/0,8599,1822455,00.html
1496
7kpe5 - AskReddit - 3467
/r/AskReddit/comments/7kpe5/i_am_constantly_wondering_what_everyone_on_here/
https://www.reddit.com/r/AskReddit/comments/7kpe5/i_am_constantly_wondering_what_everyone_on_here/
515
675oj - reddit.com - 2039
/r/reddit.com/comments/675oj/post_the_funniest_joke_you_know_upmod_the_best_one/
https://www.reddit.com/r/reddit.com/comments/675oj/post_the_funniest_joke_you_know_upmod_the_best_one/
1098
7beo2 - politics - 1934
/r/politics/comments/7beo2/obama_wins_the_presidency/
https://www.reddit.com/r/politics/comments/7beo2/obama_wins_the_presidency/
8538
6jbc0 - reddit.com - 1808
/r/reddit.com/comments/6jbc0/voice_a_true_yet_unpopular_fact_in_the

Top 10 objava po broju komentara:
1. id: 6nz1k - subreddit: science - comments: 33329  
   permalink: /r/science/comments/6nz1k/got_six_weeks_try_the_hundred_push_ups_training/  
   url: http://hundredpushups.com  
   score: 1621  
2. id: 78n1v - subreddit: WTF - comments: 3657  
   permalink: /r/WTF/comments/78n1v/a_black_community_in_oh_goes_50_years_without/  
   url: http://www.time.com/time/nation/article/0,8599,1822455,00.html  
   score: 1496  
3. id: 7kpe5 - subreddit: AskReddit - comments: 3467  
   permalink: /r/AskReddit/comments/7kpe5/i_am_constantly_wondering_what_everyone_on_here/  
   url: https://www.reddit.com/r/AskReddit/comments/7kpe5/i_am_constantly_wondering_what_everyone_on_here/  
   score: 515  
4. id: 675oj - subreddit: reddit.com - comments: 2039  
   permalink: /r/reddit.com/comments/675oj/post_the_funniest_joke_you_know_upmod_the_best_one/  
   url: https://www.reddit.com/r/reddit.com/comments/675oj/post_the_funniest_joke_you_know_upmod_the_best_one/  
   score: 1098  
5. id: 7beo2 - subreddit: politics - comments: 1934  
   permalink: /r/politics/comments/7beo2/obama_wins_the_presidency/  
   url: https://www.reddit.com/r/politics/comments/7beo2/obama_wins_the_presidency/  
   score: 8538  
6. id: 6jbc0 - subreddit: reddit.com - comments: 1808  
   permalink: /r/reddit.com/comments/6jbc0/voice_a_true_yet_unpopular_fact_in_the_comments/  
   url: https://www.reddit.com/r/reddit.com/comments/6jbc0/voice_a_true_yet_unpopular_fact_in_the_comments/  
   score: 274  
7. id: 6w7e6 - subreddit: reddit.com - comments: 1724  
   permalink: /r/reddit.com/comments/6w7e6/just_out_of_curiosity_do_you_have_an_opinion_that/  
   url: https://www.reddit.com/r/reddit.com/comments/6w7e6/just_out_of_curiosity_do_you_have_an_opinion_that/  
   score: 258  
8. id: 6yn6t - subreddit: politics - comments: 1668  
   permalink: /r/politics/comments/6yn6t/its_official_sarah_palin_is_mccains_vp_pick/  
   url: http://www.cnbc.com/id/26454655?  
   score: 1073  
9. id: 7bj3f - subreddit: reddit.com - comments: 1605  
   permalink: /r/reddit.com/comments/7bj3f/who_else_disgusted_with_california_this_morning/  
   url: http://www.latimes.com/news/local/la-me-gaymarriage5-2008nov05,0,1545381.story  
   score: 5992  
10. id: 7d7jm - subreddit: reddit.com - comments: 1505  
    permalink: /r/reddit.com/comments/7d7jm/how_many_redditors_dont_play_wow/  
    url: https://www.reddit.com/r/reddit.com/comments/7d7jm/how_many_redditors_dont_play_wow/  
    score: 7492  

In [30]:
print("\nTop10 submissions by score:")
for elem in sorted(submissionDict.items(), key = lambda item: int(item[1]["score"]), reverse=True)[:10]:
    print(str(elem[0]) + " - " + str(elem[1]["subreddit"]) + " - " + str(elem[1]["score"]))
    if elem[1]["over18"] == "False":
        print(str(elem[1]["permalink"]))
        print(str(elem[1]["url"]))


Top10 submissions by score:
7beo2 - politics - 8538
/r/politics/comments/7beo2/obama_wins_the_presidency/
https://www.reddit.com/r/politics/comments/7beo2/obama_wins_the_presidency/
7d7jm - reddit.com - 7492
/r/reddit.com/comments/7d7jm/how_many_redditors_dont_play_wow/
https://www.reddit.com/r/reddit.com/comments/7d7jm/how_many_redditors_dont_play_wow/
6x9p1 - reddit.com - 7192
/r/reddit.com/comments/6x9p1/for_every_upvote_i_will_donate_025_to_the_julie/
http://www.google.com/search?q=julie+amero
6rtav - reddit.com - 7137
/r/reddit.com/comments/6rtav/for_every_upvote_ill_donate_10_cents_to_darfur/
https://secure.ga6.org/08/savedarfurcoalition
7bj3f - reddit.com - 5992
/r/reddit.com/comments/7bj3f/who_else_disgusted_with_california_this_morning/
http://www.latimes.com/news/local/la-me-gaymarriage5-2008nov05,0,1545381.story
6v5jy - technology - 5476
/r/technology/comments/6v5jy/26_year_old_using_youtube_to_help_find_a_bone/
http://projectmichelle.com/
7iyjn - reddit.com - 5396
/r/reddi

Top 10 objava po oceni:
1. id: 7beo2 - subreddit: politics - score: 8538  
   permalink: /r/politics/comments/7beo2/obama_wins_the_presidency/  
   url: https://www.reddit.com/r/politics/comments/7beo2/obama_wins_the_presidency/  
2. id: 7d7jm - subreddit: reddit.com - score: 7492  
   permalink: /r/reddit.com/comments/7d7jm/how_many_redditors_dont_play_wow/  
   url: https://www.reddit.com/r/reddit.com/comments/7d7jm/how_many_redditors_dont_play_wow/  
3. id: 6x9p1 - subreddit: reddit.com - score: 7192  
   permalink: /r/reddit.com/comments/6x9p1/for_every_upvote_i_will_donate_025_to_the_julie/  
   url: http://www.google.com/search?q=julie+amero  
4. id: 6rtav - subreddit: reddit.com - score: 7137  
   permalink: /r/reddit.com/comments/6rtav/for_every_upvote_ill_donate_10_cents_to_darfur/  
   url: https://secure.ga6.org/08/savedarfurcoalition  
5. id: 7bj3f - subreddit: reddit.com - score: 5992  
   permalink: /r/reddit.com/comments/7bj3f/who_else_disgusted_with_california_this_morning/  
   url: http://www.latimes.com/news/local/la-me-gaymarriage5-2008nov05,0,1545381.story  
6. id: 6v5jy - subreddit: technology - score: 5476  
   permalink: /r/technology/comments/6v5jy/26_year_old_using_youtube_to_help_find_a_bone/  
   url: http://projectmichelle.com/  
7. id: 7iyjn - subreddit: reddit.com - score: 5396  
   permalink: /r/reddit.com/comments/7iyjn/vote_up_if_you_would_rather_bail_out_npr_for_30/  
   url: http://www.npr.org/templates/story/story.php?storyId=98098442&amp;ft=1&amp;f=1001  
8. id: 6rafh - subreddit: reddit.com - score: 5204  
   permalink: /r/reddit.com/comments/6rafh/vote_up_if_youre_not_getting_an_iphone_today/  
   url: https://www.reddit.com/r/reddit.com/comments/6rafh/vote_up_if_youre_not_getting_an_iphone_today/  
9. id: 74qga - subreddit: politics - score: 5078  
   permalink: /r/politics/comments/74qga/if_you_can_name_one_news_source_you_read_vote_up/  
   url: https://www.reddit.com/r/politics/comments/74qga/if_you_can_name_one_news_source_you_read_vote_up/  
10. id: 70b2f - subreddit: politics - score: 4957  
    permalink: /r/politics/comments/70b2f/is_there_anyone_else_out_there_that_wants_to_see/  
    url: https://www.reddit.com/r/politics/comments/70b2f/is_there_anyone_else_out_there_that_wants_to_see/  